src/fastcs/backends/epics/ioc.py

```diff
@@ -112,13 +114,34 @@ def _create_and_link_attribute_pvs(pv_prefix: str, mapping: Mapping) -> None:
         for attr_name, attribute in single_mapping.attributes.items():
             pv_name = attr_name.title().replace("_", "")
             _pv_prefix = ":".join([pv_prefix] + path)
+            full_pv_name_length = len(f"{_pv_prefix}:{pv_name}")
+
+            if full_pv_name_length > EPICS_MAX_NAME_LENGTH:
+                attribute.enabled = False
+                print(
+                    f"Not creating PV for {attr_name} for controller"
+                    f" {single_mapping.controller.path} as full name would exceed"
+                    f" {EPICS_MAX_NAME_LENGTH} characters"
+                )
+                continue
```

Comment:
I am not sure why we need the enabled flag. It just needs to no call `_create_and_link_*_pv`, right? What does the enabled flag add?

Reply:
We need some mechanism to tell EpicsGUI.create_gui() not to create controls for PVs that don't get created. There are other options like setting the controller attribute to None (or using delattr, but this didn't work well with pytest), anything that prevents the attribute from being provided by _get_single_mapping.

Reply:
Ah, for the GUI OK. Yeah, I am not sure it makes sense that the PV and widget creation are separated like this because they need to match. I think this is fine. I would like to revisit the overall logic of this at some point.